src/router.rs

```diff
@@ -13,6 +13,7 @@ use path_table::{PathTable, RouteMatch};
 pub struct Router<Data> {
     table: PathTable<ResourceData<Data>>,
     middleware_base: Vec<Arc<dyn Middleware<Data> + Send + Sync>>,
+    default_handler: Arc<DefaultHandler<Data>>,
```

Comment:
I've gone ahead and removed the `DefaultHandler` struct, you're right it is totally unnecessary. I'm not sure if I can remove the `Arc` trait as I am cloning it [here](https://github.com/rust-net-web/tide/pull/64/files#diff-fc414f4cdcff789e35bdc110eeed6289R186}. Let me know if I'm missing something though.

Reply:
Well, I think the default handler only makes sense when it's in the top-level router. It would be nice if it lives in `App` or somewhere, but I don't have any concrete idea now -- yeah, we might still need `Arc` in whatever approach we take.

Reply:
Yeah, I’m with you there. Having it in the sub router is kind of why I feel a little hacky with this PR. Could we potentially create a NestedRouter struct for the nest fn ?

Reply:
Hmm, how about this?

* Move `default_handler` to `App`. This will be threaded into `Server` (you'll need to wrap this with an `Arc`).
* Pass a reference to `default_handler` when you `route`.